docs/proposals/parquet-storage.md

```diff
@@ -0,0 +1,137 @@
+
+- Data organized by columns rather than rows, reduces number of requests to object storage as only limited IO is required to fetch the whole column
+- Rich file metadata and index, no local state like index header required to query the data, making it stateless
+- Advanced compression techniques reduce storage costs and improve query performance
+- Parallel processing friendly using Parquet Row Group
+
+There are other benefits of Parquet formats, but they are not directly related to the proposal:
+
+- Wide ecosystem and tooling support
+- Column pruning opportunity using projection pushdown
+
+## Out of Scope
+
+- Allow Ingester and Compactor to create Parquet files instead of TSDB blocks directly. This could be in the future roadmap but this proposal only focuses on converting and querying Parquet files.
+
+## Proposed Design
+
+### Components
+
+There are 2 new Cortex components/modules introduced in this design.
+
+#### 1. Parquet Converter
+
+Parquet Converter is a new component that converts TSDB blocks on object store to Parquet file format.
+
+It is similar to compactor, however, it only converts single block. The converted Parquet files will be stored in the same TSDB block folder so that the lifecycle of Parquet file will be managed together with the block.
+
+Only certain blocks can be configured to convert to Parquet file and it can be block duration based, for example we only convert if block duration is >= 12h.
+
+#### 2. Parquet Queryable
+
+Similar to the existing `distributorQueryable` and `blockStorageQueryable`, Parquet queryable is a queryable implementation which allows Cortex to query parquet files and can be used in both Cortex Querier and Ruler.
+
+If Parquet queryable is enabled, block storage queryable will be disabled and Cortex querier will not query Store Gateway anymore. `distributorQueryable` remains unchanged so it still queries Ingesters.
+
+Parquet queryable uses bucket index to discovers  parquet files in object storage. The bucket index is the same as the existing TSDB bucket index file, but using a different name `bucket-index-parquet.json.gz`. It is updated periodically by Cortex Compactor/Parquet Converter if parquet storage is enabled.
+
+Cortex querier remains a stateless component when Parquet queryable is enabled.
+
+### Architecture
+
+```
+┌──────────┐    ┌─────────────┐    ┌──────────────┐
+│ Ingester │───>│   TSDB      │───>│   Parquet    │
+└──────────┘    │   Blocks    │    │  Converter   │
+                └─────────────┘    └──────────────┘
+                                          │
+                                          v
+┌──────────┐    ┌─────────────┐    ┌──────────────┐
+│  Query   │───>│   Parquet   │───>│    Parquet   │
+│ Frontend │    │   Querier   │    │    Files     │
+└──────────┘    └─────────────┘    └──────────────┘
+```
+
+### Data Format
+
+Following the current design of Cortex, each Parquet file contains at most 1 day of data.
```

Comment:
I don't know nearly enough about the format, but I'm missing some more justification / explanation why 1 day, beyond current design.

Reply:
In Cortex, largest block we have is 1 day. It is configurable though.
I can remove this sentence. In short, Parquet file should have the same duration as of TSDB block as it is converted from TSDB blocks